templates/terraform_setup.yml

```diff
@@ -17,44 +17,123 @@ parameters:
+
+    base64_idrsa=$(base64 -w 0 ~/.ssh/id_rsa)
+    echo "##vso[task.setvariable variable=secret.SSH_ID_RSA_BASE64;issecret=true]${base64_idrsa}"
+    ssk_key_pub=$(cat ~/.ssh/id_rsa.pub)
+    echo "##vso[task.setvariable variable=secret.SSH_KEY_PUBLIC;issecret=true]${ssk_key_pub}"
+
+    new_node_count=$(( ${{ parameters.original_node_count }} + ${{ parameters.scale_up_node_count }} ))
+
+    sed -i -e "s/<id_rsa>/$base64_idrsa/" \
+      -e "s/<region>/$REGION/" \
+      -e "s/<avere_admin_pw>/$AVERE_ADMIN_PW/" \
+      -e "s/<sub_id>/$ARM_SUBSCRIPTION_ID/" \
+      -e "s/<arm_client_id>/$ARM_CLIENT_ID/" \
+      -e "s:<arm_client_scrt>:$ARM_CLIENT_SECRET:" \
+      -e "s/<arm_tenant_id>/$ARM_TENANT_ID/" \
+      -e "s/<resource_group_name>/$RESOURCE_GROUP_NAME/" \
+      -e "s/<vnet_name>/$VNET/" \
+      -e "s/<terraform_file_config>/$TF_VFXT_CONFIG/" \
+      -e "s,<ssh_key>,$(cat ~/.ssh/id_rsa.pub)," \
+      -e "s/<vm_admin_user>/azureuser/" \
+      -e "s/<controller_image>/${{ parameters.controller_image }}/" \
+      -e "s/<vfxt_image>/${{ parameters.image }}/" \
+      -e "s/<original_node_count>/${{ parameters.original_node_count }}/" \
+      -e "s/<new_node_count>/$new_node_count/" \
+      -e "s/<vdbench_config>/${{ parameters.vdbench_config }}/" \
+      -e "s,<vdbench_download_url>,${VDBENCH_URL}," \
+      ${cloud_init_file}
+    check_rc.sh $? "Preparing cloud-init file (sed)" || exit
+
+    # Deploy the client VMs.
+    az vm create -g ${RESOURCE_GROUP_NAME} \
+      -n ${VM_NAME}  \
+      --admin-username azureuser \
+      --vnet-name ${VNET} \
+      --subnet ${SUBNET} \
+      --image 'Canonical:0001-com-ubuntu-server-focal:20_04-lts:latest' \
+      --public-ip-address "" \
+      --size Standard_DS2_v2 \
+      --nsg "" \
+      --custom-data ${BUILD_SOURCESDIRECTORY}/templates/cloud_init/cloud_init.client.yml
+
+    check_rc.sh $? "VM Create" || exit
+
+    cmdlog=/tmp/vm_runcmd.log
+    az vm run-command invoke                  \
+      --resource-group ${RESOURCE_GROUP_NAME} \
+      --name ${VM_NAME}                      \
+      --command-id RunShellScript             \
+      --scripts "cloud-init status --wait"    \
+      --query 'value[0].message' --output tsv | tee ${cmdlog}
+    check_rc.sh ${PIPESTATUS[0]} "Wait for the VM's cloud-init to complete" || exit 0
+    grep -q 'status: done' ${cmdlog}
+    check_rc.sh $? "Wait for cloud-init completion. Create vfxt"
+
+    az vm run-command invoke                          \
+      --command-id     RunShellScript                 \
+      --name           ${VM_NAME}                     \
+      --resource-group ${RESOURCE_GROUP_NAME}         \
+      --scripts "tail -100 /var/log/cloud-init-output.log" \
+      --query 'value[0].message' --output tsv | tee /tmp/cloud-init_output.log
+
+    grep -q 'DONE: Cloud-Init COMPLETE' /tmp/cloud-init_output.log
+    check_rc.sh $? "Cloud-init status check" || exit
+
+    sleep 20
```

Comment:
If the sleep wasnt there it kept failing on the next call that run-cmd was still going on even though it wasn't since cloud init came back successful. . Unsure what else to add a check for since its already checking the cloud init being complete.

Reply:
Sounds like an `az vm run-command invoke` limitation and this is a cooldown.
```suggestion
    # Cooldown for "az vm run-command invoke" before the next one.
    # This solves a corner/race case when the command is run back-to-back.
    sleep 20
```